### PR TITLE
Feat/#307 form consolidate

### PR DESCRIPTION
**Pull request by ciaranschutte**
**Description of changes**

Make all forms use 'useFormHook'
Remove "sections" functionality; add user modal now uses a map to keep track of individual forms

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [ ] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: client/components/pages/submission-system/create-program/CreateProgramForm.js [hunk 1]
```diff
@@ -105,69 +92,57 @@ const createUpdateProgramInput = formData => ({
...
-  const [validationErrors, setValidationErrors] = React.useState({});
+  const { errors, data, setData, validateField, validateForm, touched, hasErrors } = useFormHook({
+    initialFields: formData,
+    schema: programSchema,
+  });
 
-  const programSchema = isEditing ? updateProgramSchema : createProgramSchema;
+  // TODO: ugly as sin
+  const validationErrors = errors[0];
+  const form = data[0];
```
Comment (ciaranschutte) [inline client/components/pages/submission-system/create-program/CreateProgramForm.js]: @hlminh2000 @joneubank any thoughts on a better way to do this?
Use case of having the forms in the format of `[{name: 'John', age: 45}, {name: 'Pat', age: 54}]` is `AddUser` needs multiple sections of users to all be the same overall form data.

**Comment by ciaranschutte** [comment]
@hlminh2000 removed "sections" of form hook
used the individual form map as you suggested

@joneubank @hlminh2000 How do we want to merge this because there could be breaking changes? Should we wait until the form integration tests are merged and then we can make sure it's working as it should?

**Comment by joneubank** [comment]
RE: merging... i'm just going to run it locally and check the create and edit program form variants.

File: client/global/hooks/useFormHook.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+};
+
+/**
+ * Supports single forms only, {} not [{},{}]
+ */
+const useFormHook = ({
+  initialFields,
+  schema: formSchema,
+}: {
+  initialFields: {},
```
Comment (hlminh2000) [inline client/global/hooks/useFormHook.js]: should this be `{[k:string]: any}`?

File: client/global/hooks/useFormHook.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  schema: {
+    validate: typeof yup.object,
+  },
+}) => {
+  let initErrors = {};
```
Comment (hlminh2000) [inline client/global/hooks/useFormHook.js]: should this be `{[k:string]: Error}` or whatever we expect the error type to be?

Comment (ciaranschutte) [reply client/global/hooks/useFormHook.js]: changed to {[k:string: string}
Error is some type from node

**Comment by ciaranschutte** [comment]
> if it's not too much work, can we try flowing the `components/pages/submission-system/create-program/CreateProgramForm.js ` file? :) Don't want to block development though, so can go in separate PR if we do it :)

i'll do this in separate PR

**Comment by joneubank** [comment]
Form is submitting same as before now using the form hook, but just as before it always shows an error toast (even on success). Do you want to look into why or open this as a separate bug? I'm good with either.

**Comment by hlminh2000** [comment]
> Form is submitting same as before now using the form hook, but just as before it always shows an error toast (even on success). Do you want to look into why or open this as a separate bug? I'm good with either.

I think I should look into that as it was my ticket originally to do the toast :)

**Pull request merged**